src/nextStepCliText.ts

```diff
@@ -79,97 +88,145 @@ async function outputLocalSetupCommandsIfProjectsNotAlreadyConfiguredAsLocal(
-    return `\n\nSetup target projects as local dependencies with ${context} using:`;
+  const argJoin = (arg: string[]) => {
+    return arg.join(' ');
   };
 
-  const commandsToRun = yarnCommandsIfYarnLockPresent.fold(
-    npmCommandsIfPackageLockPresent.map(npmCommandsToRun => {
-      return `${getCommandHeaderString('npm')}\n\t${npmCommandsToRun.join('\n\t')}`;
-    }),
-    yarnCommandsToRun => {
-      return some(
-        npmCommandsIfPackageLockPresent.fold(`${getCommandHeaderString('yarn')}\n\t${yarnCommandsToRun.join('\n\t')}`, npmCommandsToRun => {
-          return `${getCommandHeaderString('both')}\n\t${yarnCommandsToRun.join('\n\t')}\n  and/or\n\t${npmCommandsToRun.join('\n\t')}`;
-        })
-      );
+  const commandForPaths = (command: string, maybePaths: Option<string[]>): string => {
+    return maybePaths.fold('', paths => `\t${command} ${argJoin(paths)}\n`);
+  };
+
+  const thisCommand = cliConstants.commandName;
+  const targetArg = cliConstants.targetProjectArg;
+  const watch = cliConstants.watchCommandArg;
+
+  type StringProducer = () => string;
+  type OutputProvider = string | StringProducer | StringProducer[];
+  type OutputSpecification = OutputProvider | OutputOptions<OutputProvider>;
+
+  const outputSpec: OutputSpecification[] = [
+    `
+
+Set up target projects as local dependencies with `,
+    { npm: 'npm', npmOrYarn: ' or ', yarnClassic: 'yarn', yarnBerry: 'yarn' },
+    ` using:
+`,
+    {
+      npm: [() => commandForPaths('npm install', dependencyPaths), () => commandForPaths('npm install -D', devDependencyPaths)],
+      npmOrYarn: '  and/or\n',
+      yarnClassic: [
+        () => commandForPaths('yarn add', dependencyPaths),
+        () => commandForPaths('yarn add -D', devDependencyPaths),
+        () => '\tyarn install --check-files\n',
+      ],
+      yarnBerry: [
+        () => commandForPaths('yarn add', dependencyReferences),
+        () => commandForPaths('yarn add -D', devDependencyReferences),
+        () => '\tyarn install\n',
+      ],
+    },
+    `
+`,
+    {
+      post: () =>
+        `To get updated changes from target projects, run this command again.
+\t${thisCommand} --${targetArg} ${argJoin(targetProjectPaths)}
+  or watch continually
+\t${thisCommand} ${watch} --${targetArg} ${argJoin(targetProjectPaths)}
+
+`,
+    },
+  ];
+
+  // @ts-ignore noImplicitAny - can't typecheck this recursive return type
+  // Option<string> | (Option<string> | Option<string>[])[] | ...
+  const evaluator = (output?: OutputProvider | OutputOptions<OutputProvider>) => {
+    if (typeof output === 'string') {
+      return some(output);
+    } else if (typeof output === 'function') {
+      return evaluator(output());
+    } else if (Array.isArray(output)) {
+      return output.map(o => evaluator(o));
+    } else if (output && typeof output === 'object') {
+      return Object.keys(output)
+        .sort()
+        .map(opt => (shouldEmit[opt] ? evaluator(output[opt]) : none));
```

Comment:
Also a bit weird that `shouldEmit` is pulled from the local context but everything else is passed in. I know its written this way in order to use `map` later but wondering if we should curry this function instead.

Reply:
I definitely cheated on the functional programming in a few places... I wasn't sure how to use fp-ts for this.  Is there a utility for currying, or do we just do it manually with nested functions?

Reply:
This went away when I pulled out object support, but let me know if there are any other trouble spots.